### DISCUSSION TITLE
Can I debug MAUI preview 12 app on physical iOS device?

### DISCUSSION THREAD

**Discussion by lsvdev** [dotnet/maui#4406] [Q&A]
Hello!

Could anyone clarify, if it is possible to debug MAUI (Preview 12) app on physical iOS device using Visual Studio for Mac 2022 Pre (Preview 5) ? 

Now I see this error in log: `"ApplicationVerificationFailed: Failed to verify code signature ......  0xe800801c (No code signature found.)"`. 
And the following message appears: 
`Visual Studio cannot start the application automatically because it was signed with the Distribution provisioning profile. To launch an application, tap the corresponding icon on your device.`

Thanks.

**Comment by dmccolloughOneGas** [comment]
I would suggest that it's not possible. VS for Mac 2022 Preview does not yet support Maui.

**Reply by lsvdev** [reply]
And it is not possible with .Net CLI either? I mean not debug, but just install and launch.

**Reply by tivtag** [reply]
After quite some tinkering around I could launch the default project on a phsyical device using a command line build, not sure about debugging.

csproj:
```
    <PropertyGroup>
	<CodesignKey>YOUR_KEY</CodesignKey>
        <CodesignProvision>YOUR_PROVISIONING_PROFILE_UUID</CodesignProvision>
	<ProvisioningType>manual</ProvisioningType> 
    </PropertyGroup>

    <PropertyGroup Condition="$(TargetFramework.Contains('-ios'))">
        <RuntimeIdentifier>ios-arm64</RuntimeIdentifier>
    </PropertyGroup>
```

terminal:

dotnet build -t:Run -f net6.0-ios -p:_DeviceName=YOUR_DEVICENAME

**Reply by lsvdev** [reply]
Thanks in advance @tivtag ! 
I will give it a try and report the result here.

**Reply by lsvdev** [reply]
@tivtag That really did the trick! Magic!
Thank you so much!

I still can't debug, but my MAUI app now runs on an iPhone.

**Reply by tivtag** [reply]
@lsvdev You are welcome! Maybe you have an idea about my question: https://github.com/dotnet/maui/discussions/4438 :-)

**Reply by lsvdev** [reply]
@tivtag I wish I could help you with that, but I'm completely a newbie to this area ))

**Reply by jsuarezruiz** [reply]
For now, you require to modify the csproj to include the code sign information and use: `dotnet build -t:Run -f net6.0-ios -p:_DeviceName=DeviceName`

**Reply by lsvdev** [reply]
@jsuarezruiz Thanks!
